sycl/doc/extensions/ExplicitSIMD/dpcpp-explicit-simd.md

```diff
@@ -0,0 +1,731 @@
+```
+<br>
+
+### Private Global Variables.
+
+Explicit SIMD extenstion supports "private global" variables - file scope
+variables in private address space (similar to thread-local variables on host).
+These variables have 1 copy per work-item (which maps to a single SIMD thread in
+ESP) and are visible to all functions in the translation unit. Conceptually they
+map to SPIRV variable with private storage class. Private globals can be bound
+to a specific byte offset within the GRF. To mark a file scope variable as
+private global, the `INTEL_GPU_PRIVATE` attribute macro is used,
+`INTEL_GPU_REGISTER` is used to bind it the register file:
+
+```cpp
+INTEL_GPU_PRIVATE INTEL_GPU_REGISTER(32) simd<int, 16> vc;
+```
+<br>
+
+## Examples
+### Vector addition (USM)
+```cpp
+#include <iostream>
+#include <CL/sycl.hpp>
+#include <sycl_explicit_simd.hpp>
+
+using namespace cl::sycl;
+
+int main(void) {
+  constexpr unsigned Size = 128;
+  constexpr unsigned VL = 32;
+  constexpr unsigned GroupSize = 2;
+
+  queue q;
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+  auto ctxt = q.get_context();
+  float* A = static_cast<float*>(malloc_shared(Size*sizeof(float), dev, ctxt));
+  float* B = static_cast<float*>(malloc_shared(Size*sizeof(float), dev, ctxt));
+  float* C = static_cast<float*>(malloc_shared(Size*sizeof(float), dev, ctxt));
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = B[i] = i;
+  }
+
+  // iteration space
+  nd_range<1> Range(range<1>(Size/VL), range<1>(GroupSize));
+
+  auto e = q.submit([&](handler &cgh) {
+    cgh.parallel_for<class Test>(
+      Range, [=](nd_item<1> i) EXPLICIT_SIMD {
+
+      auto offset = i.get_global_id(0) * VL;
+      sycl::intel::gpu<float, VL> va = sycl::intel::gpu::flat_block_load<float, VL>(A + offset);
+      sycl::intel::gpu<float, VL> vb = sycl::intel::gpu::flat_block_load<float, VL>(B + offset);
+      sycl::intel::gpu<float, VL> vc = va + vb;
+      sycl::intel::gpu::flat_block_store<float, VL>(C + offset, vc);
+    });
+  });
+  e.wait();
+  int err_cnt = 0;
+
+  for (unsigned i = 0; i < Size; ++i) {
+    if (A[i] + B[i] != C[i]) {
+      if (++err_cnt < 10) {
+        std::cout << "failed at index " << i << ", " << C[i] << " != " << A[i]
+          << " + " << B[i] << "\n";
+      }
+    }
+  }
+  if (err_cnt > 0) {
+    std::cout << "  pass rate: " << ((float)(Size - err_cnt) / (float)Size)*100.0f <<
+      "% (" << (Size - err_cnt) << "/" << Size << ")\n";
+  }
+
+  std::cout << (err_cnt > 0 ? "FAILED\n" : "Passed\n");
+  return err_cnt > 0 ? 1 : 0;
+}
+```
+
+### Open questions
+- Vectorization controls (e.g. enforcing vector length generated by the
+  compiler) for vector/matrix operations.
+- Enabling loop vectorizer on inner loops in a Explicit SIMD kernel or function.
+
+### TODOs
+
+- Design interoperability with SIMT context - via `subgroup::invoke()`
```

Comment:
```suggestion
- Design interoperability with SPMD context
```

Same reason for SIMT => SPMD change as before.  There isn't a public proposal for this functionality yet and so I don't think we should give it a name (yet).